corehq/apps/locations/forms.py

```diff
@@ -241,6 +246,12 @@ def clean_site_code(self):
     def clean_location_type(self):
         loc_type = self.cleaned_data['location_type']
 
+        try:
+            loc_type_obj = LocationType.objects.get(Q(code=loc_type)|Q(name=loc_type))
```

Comment:
This one I think does make sense to check either - this form is used in bulk upload as well, which supports specifying location type either by name or code.

I have a [backlog tech-debt item](https://trello.com/c/664aSeWl/8-not-related-to-migration-make-sense-of-location-type-code-and-location-type-name-we-frequently-primarily-use-location-type-name-) to make sense of this all